test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/KubernetesMockServerTestResource.java

```diff
@@ -30,6 +38,10 @@
 
         configureMockServer(mockServer);
 
+        if (Boolean.getBoolean("quarkus.kubernetes-client.test.defaultTypes")) {
```

Comment:
This will only if `quarkus.kubernetes-client.test.defaultTypes` is a system property.
Best use `ConfigProvider.getConfig()`

Reply:
I've changed to use ConfigProvider, renamed defaultTypes to default-types and added a bit more test.